File: Programmers/Weekly/p83201.py
#LEVEL1
#코딩테스트 연습 > 위클리 챌린지 > 2주차 > 상호 평가
#https://programmers.co.kr/learn/courses/30/lessons/83201

def solution(scores):
    answer = ''
    size = len(scores)
    self_score = [0]*size
    score = [[] for _ in range(size)]
    
    for i in range(size):
        for j in range(size):
            score[i]+=[scores[j][i]]
        self_score[i] = score[i][i]
    
    for i in range(size):
        i_sum = sum(score[i])
        i_len = len(score[i])
        if self_score[i] == max(score[i]) or self_score[i] == min(score[i]):
            count=0
            for j in range(size):
                if score[i][j]==self_score[i]:
                    count+=1
            if count==1:
                i_sum-=self_score[i]
                i_len-=1
        answer+="FFFFFDDCBAA"[int(i_sum/i_len//10)]
    
    return answer

File: PS/programmers/Weekly/p83201.c
//LEVEL1
//코딩테스트 연습 > 위클리 챌린지 > 2주차 > 상호 평가
//https://programmers.co.kr/learn/courses/30/lessons/83201

#include <stdio.h>

char* solution(int** sc, size_t size, size_t dummy) {
    char* answer = (char*)malloc(size+1);answer[size]='\0';
    int min, max, sum, flag, tmp_num, tmp_count;
    
    for(int i=0;i<size;i++){
        min=100; max=sum=flag=tmp_num=0;
        
        for(int j=0;j<size;j++){
            min = sc[j][i]>min?min:sc[j][i];
            max = sc[j][i]>max?sc[j][i]:max;
            sum += sc[j][i];
        }
        
        if(sc[i][i]==min)tmp_num=min;
        else if(sc[i][i]==max)tmp_num=max;
        
        if(tmp_num!=0){
            tmp_count=0;
            for(int j=0;j<size;j++)
                tmp_count+=sc[j][i]==tmp_num?1:0;
            if(tmp_count==1)
                sum-=tmp_num,flag=1;
        }
        answer[i]="FFFFFDDCBAA"[(sum/(size-flag))/10];
    }
    return answer;
}
